### ISSUE TITLE
initPlayer issue Capacitor 4

### ISSUE THREAD

**Issue by fegauthier** [jepiqueau/capacitor-video-player#118]
Hi guys!

I'm trying to play a mkv file into ExoPlayer.

This is my Vue code to initPlayer

```
<template>
  <div id="player" class="player">
    
  </div>
</template>

<script setup lang="ts">
import { onMounted } from 'vue';
import { CapacitorVideoPlayer } from 'capacitor-video-player';

onMounted(() => {
  CapacitorVideoPlayer.initPlayer({
    mode: 'fullscreen',
    url: "********",
    playerId: "player",
    chromecast: false
  })
})

</script>

<style lang="scss" scoped>

</style>

```

This is the log in my Android Studio console

```
V/Capacitor/Plugin: To native (Capacitor plugin): callbackId: 46002438, pluginId: CapacitorVideoPlayer, methodName: initPlayer
V/Capacitor: callback: 46002438, pluginId: CapacitorVideoPlayer, methodName: initPlayer, methodData: {"mode":"fullscreen","url":"********","playerId":"player","chromecast":false}
D/CapacitorVideoPlayer: **** isTV true ****
V/CapacitorVideoPlayer: display url: ********
V/CapacitorVideoPlayer: display subtitle: 
V/CapacitorVideoPlayer: display language: 
V/CapacitorVideoPlayer: headers: {}
V/CapacitorVideoPlayer: title: 
V/CapacitorVideoPlayer: smallTitle: 
V/CapacitorVideoPlayer: accentColor: 
V/CapacitorVideoPlayer: chromecast: false
V/CapacitorVideoPlayer: artwork: 
V/CapacitorVideoPlayer: *** calculated videoPath: ********
V/CapacitorVideoPlayer: *** calculated subTitlePath: null
V/CapacitorVideoPlayer: §§§§ createFullScreenFragment chromecast: false
V/MediaRouter: Audio routes updated: AudioRoutesInfo{ type=HDMI }, a2dp=false
V/MediaRouter: Selecting route: RouteInfo{ name=HDMI, description=null, status=null, category=RouteCategory{ name=System types=ROUTE_TYPE_LIVE_AUDIO ROUTE_TYPE_LIVE_VIDEO  groupable=false }, supportedTypes=ROUTE_TYPE_LIVE_AUDIO ROUTE_TYPE_LIVE_VIDEO , presentationDisplay=null }
I/MediaRouter: Found default route: MediaRouter.RouteInfo{ uniqueId=android/androidx.mediarouter.media.SystemMediaRouteProvider:DEFAULT_ROUTE, name=HDMI, description=null, iconUri=null, enabled=true, connectionState=0, canDisconnect=false, playbackType=0, playbackStream=3, deviceType=0, volumeHandling=1, volume=16, volumeMax=25, presentationDisplayId=-1, extras=null, settingsIntent=null, providerPackageName=android }
I/MediaRouter: Unselecting the current route because it is no longer selectable: null
D/AndroidRuntime: Shutting down VM
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.tvmatters.player, PID: 27455
    java.lang.IllegalStateException: The fully qualified name of the implementation of OptionsProvider must be provided as a metadata in the AndroidManifest.xml with key com.google.android.gms.cast.framework.OPTIONS_PROVIDER_CLASS_NAME.
        at com.google.android.gms.cast.framework.CastContext.zzg(com.google.android.gms:play-services-cast-framework@@20.1.0:6)
        at com.google.android.gms.cast.framework.CastContext.getSharedInstance(com.google.android.gms:play-services-cast-framework@@20.1.0:4)
        at com.jeep.plugin.capacitor.capacitorvideoplayer.FullscreenExoPlayerFragment.onCreateView(FullscreenExoPlayerFragment.java:217)
        at androidx.fragment.app.Fragment.performCreateView(Fragment.java:2995)
        at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:523)
        at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:261)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1840)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1758)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1701)
        at androidx.fragment.app.FragmentManager$4.run(FragmentManager.java:488)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
I/Process: Sending signal. PID: 27455 SIG: 9

```

Is there something that I missed? It seems to be an error about the Cast option? I don't need this option.

Thanks

**Comment by PhantomPainX** [comment]
Even if you do not use the Chromecast option, you must follow the cast configuration to make the plugin work correctly.

The configuration given in InitPlayer only serves to disable the button and other implementations of the cast.

**Comment by fegauthier** [comment]
> Even if you do not use the Chromecast option, you must follow the cast configuration to make the plugin work correctly.
> 
> The configuration given in InitPlayer only serves to disable the button and other implementations of the cast.

If I put the configuration for Casting, I have this error at boot.

```
W/DynamiteModule: Local module descriptor class for com.google.android.gms.cast.framework.dynamite not found.
I/DynamiteModule: Considering local module com.google.android.gms.cast.framework.dynamite:0 and remote module com.google.android.gms.cast.framework.dynamite:0
D/AndroidRuntime: Shutting down VM
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.tvmatters.player, PID: 30007
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.tvmatters.player/com.tvmatters.player.MainActivity}: java.lang.RuntimeException: com.google.android.gms.cast.framework.ModuleUnavailableException: com.google.android.gms.dynamite.DynamiteModule$LoadingException: No acceptable module com.google.android.gms.cast.framework.dynamite found. Local version is 0 and remote version is 0.
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3449)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
     Caused by: java.lang.RuntimeException: com.google.android.gms.cast.framework.ModuleUnavailableException: com.google.android.gms.dynamite.DynamiteModule$LoadingException: No acceptable module com.google.android.gms.cast.framework.dynamite found. Local version is 0 and remote version is 0.
        at com.google.android.gms.cast.framework.CastContext.getSharedInstance(com.google.android.gms:play-services-cast-framework@@21.2.0:11)
        at com.tvmatters.player.MainActivity.onCreate(MainActivity.java:12)
        at android.app.Activity.performCreate(Activity.java:8000)
        at android.app.Activity.performCreate(Activity.java:7984)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3422)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:223) 
        at android.app.ActivityThread.main(ActivityThread.java:7656) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
     Caused by: com.google.android.gms.cast.framework.ModuleUnavailableException: com.google.android.gms.dynamite.DynamiteModule$LoadingException: No acceptable module com.google.android.gms.cast.framework.dynamite found. Local version is 0 and remote version is 0.
        at com.google.android.gms.internal.cast.zzad.zzf(com.google.android.gms:play-services-cast-framework@@21.2.0:7)
        at com.google.android.gms.internal.cast.zzad.zza(com.google.android.gms:play-services-cast-framework@@21.2.0:1)
        at com.google.android.gms.cast.framework.CastContext.<init>(com.google.android.gms:play-services-cast-framework@@21.2.0:4)
        at com.google.android.gms.cast.framework.CastContext.getSharedInstance(com.google.android.gms:play-services-cast-framework@@21.2.0:10)
        at com.tvmatters.player.MainActivity.onCreate(MainActivity.java:12) 
        at android.app.Activity.performCreate(Activity.java:8000) 
        at android.app.Activity.performCreate(Activity.java:7984) 
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3422) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:223) 
        at android.app.ActivityThread.main(ActivityThread.java:7656) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
     Caused by: com.google.android.gms.dynamite.DynamiteModule$LoadingException: No acceptable module com.google.android.gms.cast.framework.dynamite found. Local version is 0 and remote version is 0.
        at com.google.android.gms.dynamite.DynamiteModule.load(com.google.android.gms:play-services-basement@@18.0.0:65)
        at com.google.android.gms.internal.cast.zzad.zzf(com.google.android.gms:play-services-cast-framework@@21.2.0:2)
        at com.google.android.gms.internal.cast.zzad.zza(com.google.android.gms:play-services-cast-framework@@21.2.0:1) 
        at com.google.android.gms.cast.framework.CastContext.<init>(com.google.android.gms:play-services-cast-framework@@21.2.0:4) 
        at com.google.android.gms.cast.framework.CastContext.getSharedInstance(com.google.android.gms:play-services-cast-framework@@21.2.0:10) 
        at com.tvmatters.player.MainActivity.onCreate(MainActivity.java:12) 
        at android.app.Activity.performCreate(Activity.java:8000) 
        at android.app.Activity.performCreate(Activity.java:7984) 
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309) 
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3422) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:223) 
        at android.app.ActivityThread.main(ActivityThread.java:7656) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
I/Process: Sending signal. PID: 30007 SIG: 9

```

**Comment by PhantomPainX** [comment]
Yes, I came to the same conclusion, it is due to Play Services. You can make a PR for this fix or wait for a next release actually.

@jepiqueau

**Comment by fegauthier** [comment]
I already fixed it. I will do a PR soon :) Thanks!

**Comment by jepiqueau** [comment]
@fegauthier waiting for the PR. I will merge it

**Comment by PhantomPainX** [comment]
I think I will wait your PR too before upload an update that fixes the disappearance of the cast button and some layout improvements.

**Comment by fegauthier** [comment]
https://github.com/jepiqueau/capacitor-video-player/pull/119

**Comment by jepiqueau** [comment]
@fegauthier @PhantomPainX PR has been merged see 4.6.2-3. Thanks for your work
